examples/2d/sprite_animation.rs

```diff
@@ -64,10 +64,10 @@ fn execute_animations(time: Res<Time>, mut query: Query<(&mut AnimationConfig, &
             if let Some(atlas) = &mut sprite.texture_atlas {
                 if atlas.index == config.last_sprite_index {
                     // ...and it IS the last frame, then we move back to the first frame and stop.
-                    atlas.index = config.first_sprite_index;
+                    atlas.advance_index();
                 } else {
                     // ...and it is NOT the last frame, then we move to the next frame...
-                    atlas.index += 1;
+                    atlas.advance_index();
```

Comment:
FWIW, this change looks subtly wrong to me. Previously the logic was:

- If the index is not the last, then increment it (inline).

Now the logic is (due to `advance_index`):

- If the index is not the last, then increment it (inside `advance_index`).
- (Inside `advance_index`) If the index is more than the last set it to the index of the first.

Even if this particular change is the same as before (I didn't completely check because it looks subtly wrong and it's no longer intuitive, at least to me), my main concerns are left in a comment at the end of the review.